src/global_styling/functions/math.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+/**
+ * Utility for performing math callbacks on a string with CSS units
+ * and returning a string with its unit preserved.
+ *
+ * Example usage:
+ * mathWithUnits('4px', (x) => x / 2) = '2px';
+ * mathWithUnits(euiTheme.size.xs, (x) => x + 2) = '6px';
+ */
+export const mathWithUnits = (
+  value: string | number | undefined, // Unfortunately, this is the CSSProperties[] type used for several euiTheme vars
+  callback: (x: number) => number, // Can be multiplication, division, addition, etc.
+  unit: string = '' // Optional: if a unitless number was passed in, allow specifying a unit to return
+) => {
+  if (typeof value === 'string') {
+    const regex = /(?<value>[\d.]+)(?<unit>%|[a-zA-Z]*)/;
```

Comment:
```suggestion
    const regex = /(?<value>-?[\d.]+)(?<unit>%|[a-zA-Z]*)/;
```

Optional negative sign, we add a test case for this too

Reply:
Great catch! 🙇 https://github.com/elastic/eui/pull/6231/commits/22581fe7f10ecb50df417c82f072eb7d836d761a